compiler/AST/expr.cpp

```diff
@@ -247,7 +247,7 @@ bool Expr::isStmtExpr() const {
     retval = (parent->indexGet() != this && parent->iteratorGet() != this) ? true : false;
 
   } else {
-    retval = isBlockStmt(parentExpr);
+    retval = isDirectlyUnderBlockStmt(this);
```

Comment:
In most cases, isDirectlyUnderBlockStmt(e) will check isBlockStmt(e->parentExpr) and then merely check whether e->parentExpr's forallIntents is NULL - that's the first check in astUnderFI(). So the added runtime overhead should be modest.

Once the ForallStmt node is introduced, BlockStmt will no longer have forallIntents, and the check can be reverted from isDirectlyUnderBlockStmt(e) back to isBlockStmt(e->parentExpr).